make Release mode the default build type so users automatically get the fastest configuration (#422)
LLNL/sundials: doc/arkode/guide/source/Introduction.rst

`GitHub Issues #409 <https://github.com/LLNL/sundials/issues/409>`_ and
`#249 <https://github.com/LLNL/sundials/issues/249>`_.

> The ``CMAKE_BUILD_TYPE`` defaults to ``Release`` mode now. I.e., SUNDIALS will
> be built with optimization flags by default. Previously it defaulted to empty so
> no optimization or debug flags were set.

Reviewer: ```suggestion
The ``CMAKE_BUILD_TYPE`` defaults to ``Release`` mode now i.e., SUNDIALS will be
built with optimizations enabled by default. Previously the build type was unset
by default so no optimization or debug flags were set.
```

> The ``CMAKE_BUILD_TYPE`` defaults to ``Release`` mode now i.e., SUNDIALS will be
> built with optimizations enabled by default. Previously the build type was unset
> by default so no optimization or debug flags were set.

Reviewer: ```suggestion
The ``CMAKE_BUILD_TYPE`` defaults to ``RelWithDebInfo`` mode now i.e., SUNDIALS
will be built with optimizations and debugging symbols enabled by default.
Previously the build type was unset by default so no optimization or debugging
flags were set.
```
